sympy/vector/vector.py

```diff
@@ -418,19 +418,22 @@ def _div_helper(one, other):
 
 def get_postprocessor(cls):
     def _postprocessor(expr):
-        vec_class = {Add: VectorAdd}[cls]
+        vec_class = {Add: VectorAdd, Mul: VectorMul}[cls]
         vectors = []
         for term in expr.args:
             if isinstance(term.kind, VectorKind):
                 vectors.append(term)
 
         if vec_class == VectorAdd:
-            return VectorAdd(*vectors).doit(deep=False)
+            return VectorAdd(*expr.args).doit(deep=False)
+        if vec_class == VectorMul:
+            return VectorMul(*vectors).doit(deep=False)
```

Comment:
Can this not just pass all of `expr.args`? What happens to the other args?

I think there needs to be some tests in the `test_vector.py` for what happens when calling `Add(*args)` or `Mul(*args)` with args that are vectors or scalars. These can go along with the tests for kind.